galasa-ui/src/tests/usersTable.test.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import UsersTable from '@/components/users/UsersTable';
+import { UserData } from '@/generated/galasaapi';
+import '@testing-library/jest-dom';
+
+// ------------------------------
+// Mocks
+// ------------------------------
+
+// Mock Carbon's Loading component to render a simple div.
+jest.mock('@carbon/react', () => {
+  const originalModule = jest.requireActual('@carbon/react');
+  return {
+    __esModule: true,
+    ...originalModule,
+    Loading: () => <div data-testid="loading">Loading...</div>
+  };
+});
+
+// Mock the ErrorPage component.
+jest.mock('@/app/error/page', () => {
+  function ErrorPageMock() {
+    return <div data-testid="error-page">Error Page</div>;
+  }
+  return ErrorPageMock;
+});
+
+// Mock the CSS module.
+jest.mock('@/styles/UsersList.module.css', () => ({
+  userListContainer: 'userListContainer'
+}));
+
+// ------------------------------
+// Tests
+// ------------------------------
+describe('UsersTable component', () => {
+  test('displays loading spinner while fetching data', async () => {
+    // Create a promise that never resolves to simulate the loading state.
+    const pendingPromise: Promise<UserData[]> = new Promise(() => {});
+    render(<UsersTable usersListPromise={pendingPromise} />);
+    
+    // The Loading component should be rendered.
+    expect(screen.getByTestId('loading')).toBeInTheDocument();
+  });
+
+  test('renders table with user data after promise resolves', async () => {
+    // Sample user data with both "web-ui" and "rest-api" clients.
+    const userData: UserData[] = [
+      {
+        id: '1',
+        loginId: 'user1',
+        synthetic: {
+          role: {
+            metadata: { name: 'admin' }
+          }
+        },
+        clients: [
+          {
+            clientName: 'web-ui',
+            lastLogin: new Date('2020-01-01T00:00:00Z')
+          },
+          {
+            clientName: 'rest-api',
+            lastLogin: new Date('2020-01-02T00:00:00Z')
+          }
+        ]
+      }
+    ];
+
+    const resolvedPromise: Promise<UserData[]> = Promise.resolve(userData);
+    render(<UsersTable usersListPromise={resolvedPromise} />);
+
+    // Wait until the table header appears, indicating data has been loaded.
+    await waitFor(() => {
+      expect(screen.getByText('Login ID')).toBeInTheDocument();
+    });
+
+    // Check that the user data is rendered.
+    expect(screen.getByText('user1')).toBeInTheDocument();
+    expect(screen.getByText('admin')).toBeInTheDocument();
+
+    // Verify that the formatted dates are rendered.
+    const formattedLastLogin = new Intl.DateTimeFormat('en-GB').format(new Date('2020-01-01T00:00:00Z'));
+    const formattedLastAccessTokenUse = new Intl.DateTimeFormat('en-GB').format(new Date('2020-01-02T00:00:00Z'));
```

Comment:
Maybe just use the expected DD/MM/YYYY string instead of using the same formatting logic? e.g. "02/01/2020"